### DISCUSSION TITLE
Problem with exceptions

### DISCUSSION THREAD

**Discussion by LaptosPirose** [ottowayi/pycomm3#296] [General]
My Ethernet connection is not so good and I having some problems with communication loss. I have no idea how avoid programm stop because of Time out.

Traceback (most recent call last):
  File "C:\Users\Engenharia\AppData\Local\Programs\Python\Python311\Lib\site-packages\pycomm3\socket_.py", line 65, in receive
    data = self.sock.recv(256)
           ^^^^^^^^^^^^^^^^^^^
TimeoutError: timed out

How can I do to reconnect?

**Comment by MAD-Thrustm** [comment]
Could you share the snippet of your code that encountered this error?
If it helps, I usually handle exception errors by try:... except:...
for example 
from pycomm3 import LogixDriver
from pycomm3.exceptions import CommError
    plc = LogixDriver(PLC_IP_ADDRESS)
    plc.open()
    plc.open()
    while True:
        try:
            print(plc.get_plc_name)
        except CommError as err:
            plc.close()
            print(f"CommError: {err}")
        except TimeoutError as err:
            plc.close()
            print(f"TimeoutError: {err}")
        except:
            plc.close()
            print("unknown error!")
